src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/resilience/RollbacksSpec.groovy

```diff
@@ -255,4 +257,32 @@ and at least 1 path must remain safe"
                 ]
         ]
     }
+
+    @Tidy
+    def "Flow is successfully deleted from the system even if some rule delete commands fail (no rollback for delete)"() {
+        given: "A flow"
+        def swPair = topologyHelper.switchPairs.first()
+        def flow = flowHelperV2.randomFlow(swPair)
+        flowHelperV2.addFlow(flow)
+
+        when: "Send delete request for the flow"
+        lockKeeper.shapeSwitchesTraffic([swPair.src], new TrafficControlData(1000))
+        northboundV2.deleteFlow(flow.flowId)
+
+        and: "One of the related switches does not respond"
+        def blockData = switchHelper.knockoutSwitch(swPair.src, mgmtFlManager)
+
+        then: "Flow history shows failed delete rule retry attempts but flow deletion is successful at the end"
+        wait(WAIT_OFFSET) {
+            def history = northbound.getFlowHistory(flow.flowId).last().histories
+            //egress and ingress rule on a broken switch, 3 retries each = total 6
+            assert history.count { it.details ==~ /Failed to remove the rule.*Retrying \(attempt \d+\)/ } == 6
+            assert history.last().action == DELETE_SUCCESS
+        }
+        northboundV2.getAllFlows().empty
```

Comment:
`!northboundV2.getFlowStatus(flowId)`
it allows us to not fail if some flow already exist

Reply:
done